src/components/ArticlesByTopic.jsx

```diff
@@ -0,0 +1,26 @@
+import { useLocation } from "react-router-dom";
+import ArticleBlock from "./ArticleBlock";
+import ArticleBlockSmall from "./ArticleBlockSmall";
+
+function ArticlesByTopic({ articles }) {
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+  const topic = queryParams.get("topic");
+
+  const articlesByTopic = articles.filter((article) => {
```

Comment:
although this shouldn't be here yet (!), when you come back to the ticket concerned with this, make use of the querying for topics functionality on the /articles enpoint on the backend for this if you have added it! :)